src/utils.rs

```diff
@@ -8,10 +8,24 @@ use tilejson::Bounds;
 
 use crate::source::{Query, Xyz};
 
-pub fn prettify_error<E: std::fmt::Display>(message: String) -> impl Fn(E) -> std::io::Error {
-    move |error| std::io::Error::new(std::io::ErrorKind::Other, format!("{message}: {error}"))
+#[macro_export]
+macro_rules! prettify_error {
+    ($error:ident,$info:expr,$($arg:expr),+) => {
+        {
+            std::io::Error::new(std::io::ErrorKind::Other,format!("{}: {}",
+            format!($info,$($arg),*),
+            $error))
+        }
+    };
+    ($error:ident,$info:literal) => {
+        {
+            std::io::Error::new(std::io::ErrorKind::Other,format!("{}: {}",$info,$error))
```

Comment:
If possible, we don't want to do two separate string allocations (one in each `format!`), so maybe do static magic with `concat!()`?   I am not certain this will work, but if its a literal, it should?

```suggestion
            ::std::io::Error::new(::std::io::ErrorKind::Other, format!(concat!($info, ": {}", $error))
```